queue/priority.go

```diff
@@ -102,6 +104,12 @@ func (q *priorityLocalQueue) Next(ctx context.Context) amboy.Job {
 				}()
 				continue
 			}
+			if err := q.scopes.Acquire(job.ID(), job.Scopes()); err != nil {
+				go func() {
+					defer recovery.LogStackTraceAndContinue("re-queue waiting job", job.ID())
+					q.channel <- job
```

Comment:
Why send a job in the channel if acquiring he lock fails?

Reply:
because otherwise it falls on the floor and is never acted upon.